### PR TITLE
datadog-agent: 7.38.1 -> 7.45.1

### PR DESCRIPTION
**Pull request by flurie**
###### Description of changes

https://github.com/DataDog/datadog-agent/blob/7.45.1/CHANGELOG.rst

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/networking/dd-agent/datadog-agent.nix [hunk 1]
```diff
@@ -35,7 +35,7 @@ let
     cmakeFlags = ["-DBUILD_DEMO=OFF" "-DDISABLE_PYTHON2=ON"];
   };
 
-in buildGoModule rec {
+in buildGo118Module rec {
```
Comment (flurie) [inline pkgs/tools/networking/dd-agent/datadog-agent.nix:38]: Please note that a previous PR that unpinned the version of go here broke the package due to a transitive dep.

Comment (marsam) [reply pkgs/tools/networking/dd-agent/datadog-agent.nix:38]: Can we use `buildGo119Module`? upstream seems to use go 1.19 https://github.com/DataDog/datadog-agent/blob/7.45.1/.go-version

Comment (flurie) [reply pkgs/tools/networking/dd-agent/datadog-agent.nix:38]: Unfortunately, we cannot. That results in the same issue with the transitive dep:

```
[flurie@a-m:~/code/github/flurie/nixpkgs]$ ./result/bin/agent 
panic: Something in this program imports go4.org/unsafe/assume-no-moving-gc to declare that it assumes a non-moving garbage collector, but your version of go4.org/unsafe/assume-no-moving-gc hasn't been updated to assert that it's safe against the go1.19 runtime. If you want to risk it, run with environment variable ASSUME_NO_MOVING_GC_UNSAFE_RISK_IT_WITH=go1.19 set. Notably, if go1.19 adds a moving garbage collector, this program is unsafe to use.
```

**Comment by flurie** [comment]
I've just added a separate config for the datadog process agent in the nixos module as well. It was hard-coded, and it should likely be configurable as well (for situations like this).

**Comment by dmvinson** [comment]
Hi, is this planned on being merged? The datadog agent is currently broken in the latest nixpkgs.

**Pull request merged**